plugins/Ntuplizer.cc

```diff
@@ -522,211 +553,247 @@ Ntuplizer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
-    //	}
+    if(isZtoMM) {
+    if(isVerbose) std::cout << "Do the Z->mu mu" << std::endl;
+    int m1(-1), m2(-1);
+    for(unsigned int i = 0; i < TightMuonVect.size(); i++) {
+    	for(unsigned int j = 1; j < TightMuonVect.size(); j++) {
+    	  if(i==j || TightMuonVect[i].charge() == TightMuonVect[j].charge())
+    	    {
+    	      isZtoMM = false;
+    	      continue;
+    	    }
+    	  float Zmass = (TightMuonVect[i].p4() + TightMuonVect[j].p4()).mass();
+    	  if(Zmass > 50. && Zmass < 130.)
+    	    {
+    	      m1 = i;
+    	      m2 = j;
+    	      isZtoMM = true;
+    	    }
+    	}
+    }
+    //    Build candidate
+    if(m1 >= 0 && m2 >= 0) {
+    	theZ.addDaughter(TightMuonVect.at(m1).charge() < 0 ? TightMuonVect.at(m1) : TightMuonVect.at(m2));
+    	theZ.addDaughter(TightMuonVect.at(m1).charge() < 0 ? TightMuonVect.at(m2) : TightMuonVect.at(m1));
+    	addP4.set(theZ);
+    	isZtoMM = true;
+
+    	//SF
+
+    	if(isControl && isMC && !is2016) {
+    	  float LeptonWeightUnc = 0.;
+	  LeptonWeightUp = 0.;
+	  LeptonWeightDown = 0.;
+    	  /// FIXED -> APPLYING THE SF FOR IsoMu24 NOT ANYLONGER HADRCODED <- FIXED ///
+	  if (isIsoMu24_OR_IsoTkMu24) {
+	    if (TightMuonVect.at(m1).pt() > TightMuonVect.at(m2).pt() ) {
+	      LeptonWeight     *= theMuonAnalyzer->GetMuonTriggerSFIsoMu24(TightMuonVect.at(m1));
+	      LeptonWeightUnc  += pow(theMuonAnalyzer->GetMuonTriggerSFErrorIsoMu24(MuonVect.at(m1)),2);
+	    }
+	    else {
+	      LeptonWeight     *= theMuonAnalyzer->GetMuonTriggerSFIsoMu24(TightMuonVect.at(m2));
+	      LeptonWeightUnc  += pow(theMuonAnalyzer->GetMuonTriggerSFErrorIsoMu24(MuonVect.at(m2)),2);
+	    }
+	  }// IsoMu24 trigger
+	  if (isMu50_OR_TkMu50) {
+	    if (TightMuonVect.at(m1).pt() > TightMuonVect.at(m2).pt() ) {
+	      LeptonWeight     *= theMuonAnalyzer->GetMuonTriggerSFMu50(TightMuonVect.at(m1));
+	      LeptonWeightUnc  += pow(theMuonAnalyzer->GetMuonTriggerSFErrorMu50(MuonVect.at(m1)),2);
+	    }
+	    else {
+	      LeptonWeight     *= theMuonAnalyzer->GetMuonTriggerSFMu50(TightMuonVect.at(m2));
+	      LeptonWeightUnc  += pow(theMuonAnalyzer->GetMuonTriggerSFErrorMu50(MuonVect.at(m2)),2);
+	    }
+	  }// IsoMu50 trigger
 
+	  // //removed obsolete things for now
+	  // LeptonWeight *= theMuonAnalyzer->GetMuonTrkSF(TightMuonVect.at(m1));
+	  // LeptonWeight *= theMuonAnalyzer->GetMuonTrkSF(TightMuonVect.at(m2));
+    	  // //LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonTrkSFError(MuonVect.at(m1))      ,2);
+    	  // //LeptonWeightUnc += pow(theMuonAnalyzer->GetMuonTrkSFError(MuonVect.at(m2))      ,2);
 
-    //}
-    //}
+    	  LeptonWeight *= theMuonAnalyzer->GetMuonIdSF(TightMuonVect.at(m1), 1);
+    	  LeptonWeight *= theMuonAnalyzer->GetMuonIdSF(TightMuonVect.at(m2), 1);
+    	  LeptonWeight *= theMuonAnalyzer->GetMuonIsoSF(TightMuonVect.at(m1), 1);
+    	  LeptonWeight *= theMuonAnalyzer->GetMuonIsoSF(TightMuonVect.at(m2), 1);
```

Comment:
This vector is filled with muons with tight ID (i.e. id==3) but the SF are computed for id=1. Is this intended?

Reply:
This looks wrong to me, well spotted. @MelanieEich ?

Reply:
You are right! Thanks for spotting!

Reply:
Please use here tight id = 3.

Reply:
Done!